src/gfx_layout.h

```diff
@@ -155,14 +157,25 @@ class Layouter : public std::vector<std::unique_ptr<const ParagraphLayouter::Lin
 	struct LineCacheKey {
 		FontState state_before;  ///< Font state at the beginning of the line.
 		std::string str;         ///< Source string of the line (including colour and font size codes).
+	};
+
+	struct LineCacheQuery {
+		FontState state_before;  ///< Font state at the beginning of the line.
+		std::string_view str; ///< Source string of the line (including colour and font size codes).
+	};
+
+	/** Comparator for std::map */
+	struct LineCacheCompare {
+		using is_transparent = void; ///< Enable map queries with various key types
```

Comment:
Don't understand why this is necessary?

Reply:
This is required to enable some overloads in `std::map`. [Example, 3-4](https://en.cppreference.com/w/cpp/container/map/find)